google/api_core/universe_helpers.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Helpers for universe domain."""
+
+from google.auth.exceptions import MutualTLSChannelError
+from typing import Any, Optional
+
+_DEFAULT_UNIVERSE = "googleapis.com"
+
+_mTLS_Universe_Error = MutualTLSChannelError(
+    f"mTLS is not supported in any universe other than {_DEFAULT_UNIVERSE}."
+)
+
+
+class _UniverseMismatchError(ValueError):
+    def __init__(self, client_universe, credentials_universe):
+        message = (
+            f"The configured universe domain ({client_universe}) does not match the universe domain "
+            f"found in the credentials ({credentials_universe}). "
+            "If you haven't configured the universe domain explicitly, "
+            f"`{_DEFAULT_UNIVERSE}` is the default."
+        )
+        super().__init__(message)
+
+
+def _get_universe_domain(
```

Comment:
These two methods are underscore private, but we want to make them public here, right?